components/movementsensor/adxl345/adxl345.go

```diff
@@ -41,14 +41,57 @@ var modelName = resource.NewDefaultModel("accel-adxl345")
 
 // AttrConfig is a description of how to find an ADXL345 accelerometer on the robot.
 type AttrConfig struct {
-	BoardName              string `json:"board"`
-	I2cBus                 string `json:"i2c_bus"`
-	UseAlternateI2CAddress bool   `json:"use_alternate_i2c_address,omitempty"`
+	BoardName              string                 `json:"board"`
+	I2cBus                 string                 `json:"i2c_bus"`
+	UseAlternateI2CAddress bool                   `json:"use_alternate_i2c_address,omitempty"`
+	InterruptPin1          InterruptPinAttrConfig `json:"interrupt_pin1,omitempty"`
```

Comment:
These configs don't look right

FreeFallAttrConfig looks right. TapAttrConfig should have an interrupt pin. From there add both to the AttrConfig. A lot of your code will get simplified after this.

```
type AttrConfig struct {
BoardName string ...
I2cBus string....
UseAlternateIc2CAddress bool .... //omitempty
FreeFallConfig FreeFallAttrConfig ... //omitempty
TapConfig TapAttrConfig .... //omitempty
}
```